service/src/main/java/uk/gov/hmcts/reform/fpl/handlers/NotificationHandler.java

```diff
@@ -0,0 +1,101 @@
+
+import java.util.LinkedHashMap;
+import java.util.Map;
+import java.util.Optional;
+
+import static uk.gov.hmcts.reform.fpl.CaseDefinitionConstants.CASE_TYPE;
+import static uk.gov.hmcts.reform.fpl.CaseDefinitionConstants.JURISDICTION;
+import static uk.gov.hmcts.reform.fpl.NotifyTemplates.HMCTS_COURT_SUBMISSION_TEMPLATE;
+
+@Component
+public class NotificationHandler {
+
+    private final HmctsCourtLookUpService hmctsCourtLookUpService;
+    private final LocalAuthorityService localAuthorityService;
+    private final NotificationClient notificationClient;
+    private final Logger logger = LoggerFactory.getLogger(getClass());
+    private final String uiBaseUrl;
+
+    @Autowired
+    public NotificationHandler(HmctsCourtLookUpService hmctsCourtLookUpService,
+                               NotificationClient notificationClient,
+                               LocalAuthorityService localAuthorityService,
+                               @Value("${ccd.ui.base.url}") String uiBaseUrl) {
+        this.hmctsCourtLookUpService = hmctsCourtLookUpService;
+        this.notificationClient = notificationClient;
+        this.localAuthorityService = localAuthorityService;
+        this.uiBaseUrl = uiBaseUrl;
+    }
+
+    @EventListener
+    public void sendNotificationToHmctsAdmin(SubmittedCaseEvent event) {
+        CaseDetails caseDetails = event.getCallbackRequest().getCaseDetails();
+        String localAuthorityCode = caseDetails.getData().get("caseLocalAuthority").toString();
+        Map<String, String> parameters = buildEmailData(caseDetails, localAuthorityCode);
+        String reference = caseDetails.getId().toString();
+
+        String email = hmctsCourtLookUpService.getCourt(localAuthorityCode).getEmail();
+        logger.debug("Sending email to {}", email);
+
+        try {
+            notificationClient.sendEmail(HMCTS_COURT_SUBMISSION_TEMPLATE, email, parameters, reference);
+        } catch (NotificationClientException e) {
+            logger.warn("Failed to send email to {}", email, e);
+
+            e.getHttpResult();
+            e.getMessage();
+        }
+    }
+
+    private Map<String, String> buildEmailData(CaseDetails caseDetails, String localAuthorityCode) {
+        LinkedHashMap orders =
+            Optional.ofNullable((LinkedHashMap) caseDetails.getData().get("orders")).orElse(new LinkedHashMap());
+
+        LinkedHashMap hearing =
+            Optional.ofNullable((LinkedHashMap) caseDetails.getData().get("hearing")).orElse(new LinkedHashMap());
+
+        String timeFramePresent;
+        String timeFrame = "timeFrame";
+
+        if (hearing.containsKey(timeFrame)) {
+            timeFramePresent = "Yes";
+        } else {
+            timeFramePresent = "No";
+        }
+
+        String orderType = Optional.ofNullable(orders.get("orderType"))
+            .orElse("").toString();
+        String directions = Optional.ofNullable(orders.get("directionsAndInterim")).orElse("").toString();
```

Comment:
Please cast instead of using `toString`. Applies to other places.

Using `toString` can give unexpected results and hide real issue. Cast will fail fast.